src/LandIce/evaluators/LandIce_EnthalpyResid_Def.hpp

```diff
@@ -127,6 +135,128 @@ EnthalpyResid(const Teuchos::ParameterList& p, const Teuchos::RCP<Albany::Layout
   flux_reg_beta = flux_reg_list.get<double>("beta");
 }
 
+template<typename EvalT, typename Traits, typename VelocityST, typename MeltTempST>
+KOKKOS_INLINE_FUNCTION
+void EnthalpyResid<EvalT,Traits,VelocityST,MeltTempST>::
+stabilizationInitialization(int cell, VelocityST& vmax_xy, ScalarT& vmax, ScalarT& vmax_z, 
+  MeshScalarT& diam, MeshScalarT& diam_xy, MeshScalarT& diam_z, ScalarT& wSU) const {
+  for (std::size_t qp = 0; qp < numQPs; ++qp) {
+      ScalarT w = verticalVel(cell,qp);
```

Comment:
So I think we've had issues in the past with using auto for fad types in kernels. I don't remember the exact issue unfortunately. It might worth exploring again.

Reply:
Interesting. I also remember some issue with auto. But if I remember correctly, we were using auto temporaries _as if_ they were some sort of scalar type, while in Sacado they may be expressions.

I would explore it, but maybe we should check with Eric Phipps if storing expressions for temporaries is indeed beneficial or not. I _think_ it is.

Reply:
It probably depends on whether you'll be reusing the memory or not. In this case, auto might actually make sense.